File: data/src/main/java/com/roman/kubik/spivanyklicejista/database/AppDatabase.java
package com.roman.kubik.spivanyklicejista.database;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.RoomDatabase;

import com.roman.kubik.spivanyklicejista.category.CategoryEntity;
import com.roman.kubik.spivanyklicejista.song.SongDao;
import com.roman.kubik.spivanyklicejista.song.SongEntity;

/**
 * Room database representation
 * Created by kubik on 1/14/18.
 */
@Database(entities = {SongEntity.class, CategoryEntity.class}, version = 1)
public abstract class AppDatabase extends RoomDatabase {

    public abstract SongDao songDao();

}

File: song/songsRoom/src/main/java/com/roman/kubik/songer/room/database/AppDatabase.kt
package com.roman.kubik.songer.room.database

import androidx.room.Database
import androidx.room.RoomDatabase
import com.roman.kubik.songer.room.category.CategoryEntity
import com.roman.kubik.songer.room.favourite.FavouriteDao
import com.roman.kubik.songer.room.favourite.FavouriteEntity
import com.roman.kubik.songer.room.history.HistoryDao
import com.roman.kubik.songer.room.history.HistoryEntity
import com.roman.kubik.songer.room.lastadded.LastAddedDao
import com.roman.kubik.songer.room.lastadded.LastAddedEntity
import com.roman.kubik.songer.room.song.SongDao
import com.roman.kubik.songer.room.song.SongEntity

/**
 * Room database representation
 */
@Database(entities = [SongEntity::class, CategoryEntity::class, FavouriteEntity::class, HistoryEntity::class, LastAddedEntity::class], version = 6, exportSchema = false)
abstract class AppDatabase : RoomDatabase() {

    abstract fun songDao(): SongDao

    abstract fun favouriteDao(): FavouriteDao

    abstract fun historyDao(): HistoryDao

    abstract fun lastAddedDao(): LastAddedDao
}